shell/common/shell.cc

```diff
@@ -992,11 +992,27 @@ void Shell::OnPlatformViewSetViewportMetrics(const ViewportMetrics& metrics) {
   }
 }
 
+std::set<std::string> Shell::misbehaving_message_channels_ = {};
```

Comment:
This isn't threadsafe either.  In the error case you are talking to this on multiple threads.

Reply:
I could introduce a mutex and then make this only for non-release.

Reply:
The problem with making it thread local is that it could still potentially print out very frequently in a plugin that's getting callbacks on new threads each time.

Reply:
I'm just doing this per shell. In 99% of cases there will only be one shell anyway, and if someone gets duplicate messages due to more than one shell that's probably fine.